File: c++/leetcode/1356. Sort Integers by The Number of 1 Bits.cpp
class Solution
{
public:
  vector<int> sortByBits(vector<int> &arr)
  {

    auto count_bits = [](int n) {
      int c{};
      for (c = 0; n; ++c)
        n &= n - 1;

      return c;
    };

    sort(arr.begin(), arr.end(), [&](int a, int b) {
      if (count_bits(a) != count_bits(b))
        return count_bits(a) < count_bits(b);

      return a < b;
    });

    return arr;
  }
};

File: javascript/leetcode/1356. Sort Integers by The Number of 1 Bits.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var sortByBits = function (arr) {
  const count_bits = (n) => {
    let c = 0;

    for (; n; ++c) n &= n - 1;

    return c;
  };

  arr.sort((a, b) => {
    if (count_bits(a) !== count_bits(b)) return count_bits(a) - count_bits(b);

    return a - b;
  });

  return arr;
};
